### DISCUSSION TITLE
if jedis's impl support concurrency?

### DISCUSSION THREAD

**Discussion by Lqinhao** [xetorthio/jedis#2434] [Q&A]
I am using Jedis to add values to redis value of set. And i wonder if JedisSetCommonds its self impl support concurrency access of redis connect to add values to set?

**Comment by sazzad16** [comment]
@Lqinhao Could you please rephrase your question?

**Reply by Lqinhao** [reply]
I used set as values to cache someInformation. And with java application, I use redisTemplete to add new value to my cached values.
Then I use redisTemplete.opsForSet().add() method, it's basic implement using JedisSetCommands to improve it.
So I wonder if concurrency access to add value to values of set type, would JedisSetCommands support this situation and improve it?

**Reply by sazzad16** [reply]
Please check [spring-data-redis](https://github.com/spring-projects/spring-data-redis). Thanks

**Comment by dengliming** [comment]
Not sure if I fully understand your question. Jedis is just a Redis client. We don't do concurrency control on the client side. If you want to add multi values to a `Set`. just put it in one command. e.g.  `JedisSetCommands#sAdd(byte[] key, byte[]... values)`. For other scenarios you can also use Lua scripts.
